vimeo-networking/src/main/java/com/vimeo/networking/model/VimeoAccount.java

```diff
@@ -39,81 +40,142 @@
 
     private static final long serialVersionUID = -8341071767843490585L;
-    //    private static final String TOKEN_TYPE_BEARER = "bearer";
 
+    @Nullable
     @GsonAdapterKey("access_token")
-    public String accessToken;
+    public String mAccessToken;
+
+    @Nullable
     @GsonAdapterKey("token_type")
-    public String tokenType;
+    public String mTokenType;
+
+    @Nullable
     @GsonAdapterKey("scope")
-    public String scope;
+    public String mScope;
+
+    @Nullable
     @GsonAdapterKey("user")
-    public User user;
-    private String userJSON;
+    public User mUser;
+
+    @Nullable
+    private String mUserJson;
 
-    public VimeoAccount(){
+    public VimeoAccount() {
         //constructor for stag TypeAdapter generation
     }
 
     public VimeoAccount(@Nullable String accessToken) {
-        this.accessToken = accessToken;
+        mAccessToken = accessToken;
     }
 
-    public VimeoAccount(String accessToken, String tokenType, String scope, String userJSON) {
-        if (accessToken == null || accessToken.isEmpty() || tokenType == null ||
-            tokenType.isEmpty() || scope == null || scope.isEmpty()) {
+    public VimeoAccount(@Nullable String accessToken, @NotNull String tokenType, @NotNull String scope,
+                        String userJSON) {
+        if ((accessToken != null && accessToken.isEmpty()) || tokenType == null || tokenType.isEmpty() ||
+            scope == null || scope.isEmpty()) {
             throw new AssertionError("Account can only be created with token, tokenType, scope");
         }
 
-        this.accessToken = accessToken;
-        this.tokenType = tokenType;
-        this.scope = scope;
+        mAccessToken = accessToken;
+        mTokenType = tokenType;
+        mScope = scope;
 
-        Gson gson = VimeoNetworkUtil.getGson();
+        if (userJSON != null) {
+            Gson gson = VimeoNetworkUtil.getGson();
 
-        this.user = gson.fromJson(userJSON, User.class);
+            mUser = gson.fromJson(userJSON, User.class);
+            mUserJson = userJSON;
+        }
     }
 
+    /**
+     * @return true if the access token is not empty, false if it is null or empty
+     */
     public boolean isAuthenticated() {
-        return (this.accessToken != null && !this.accessToken.isEmpty());
+        return (mAccessToken != null && !mAccessToken.isEmpty());
     }
 
+    /**
+     * @return the access (auth) token stored with this account
+     */
+    @Nullable
     public String getAccessToken() {
-        return this.accessToken;
+        return mAccessToken;
     }
 
+    /**
+     * Sets the auth token. This should only be used when there is no auth token set, or it otherwise
+     * needs to change. If changed, this {@link VimeoAccount} should be set and saved in the client
+     * for use.
+     *
+     * @param accessToken the new auth token
+     */
+    public void setAccessToken(@Nullable String accessToken) {
```

Comment:
Why is this method necessary? We have a constructor that takes an auth token. I'm wary or making this externally settable as well as allowing it to be null.

Related to my comments above